fix: LEAP-1919: update origin on editing classifications results

Gondragos opened the pull request:
### Description  
This update ensures prediction results origins are consistently updated to "prediction-changed" when user edits are made to classification results. It includes:  
- A new `updateOriginOnEdit()` method in `Regions.js` for handling origin updates.  
- Adjustments to `ToolBar` controls to better support actions like annotation duplication.  
- Integration tests verifying that predictions are updated correctly post-edit.

### Reason for Change  
Previously, user edits did not update the origin value consistently, leading to potential data inconsistencies. This fix ensures user actions are accurately reflected in prediction metadata.

### Testing  
Integration tests have been implemented to verify proper behavior under various scenarios of user editing. 

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Gondragos committed:
fix: update origin on editing classifications results

Ensure prediction origins are updated to "prediction-changed" upon user edits.
- Added `updateOriginOnEdit()` method in `Regions.js` to handle origin updates.
- Improved `ToolBar` controls to manage actions like duplicating annotations.
- Introduced integration tests to validate correct origin updates for predictions.

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14247031652)
> Successfully merged:  create mode 100644 web/libs/ui/src/shadcn.ts

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1919/classification-prediction'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14247031652

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14261255329)
> Successfully merged:  3 files changed, 73 insertions(+), 14 deletions(-)

MihajloHoma committed:
Merge branch 'develop' into 'fb-LEAP-1919/classification-prediction'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14261255329

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14261336670)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14261336670

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14312632056)
> Successfully merged:  create mode 100644 label_studio/users/product_tours/configs/show_ask_ai.yml

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1919/classification-prediction'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14312632056

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14313498706)
> Successfully merged:  2 files changed, 2 insertions(+), 2 deletions(-)

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1919/classification-prediction'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14313498706

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14333258413)
> Successfully merged:  create mode 100644 docs/themes/v2/source/images/storages/s3-trust-policy.png

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1919/classification-prediction'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14333258413

Gondragos committed:
test: add e2e test for undo functionality

Implemented a new end-to-end test to verify the correct behavior of the undo feature in Label Studio. The test ensures that annotations revert to their predicted state after multiple undo actions. Added Hotkeys helper to simulate undo operations.

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14339930628)
> Successfully merged: Already up to date.

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14357953765)
> Successfully merged:  384 files changed, 56 insertions(+), 4785 deletions(-)

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1919/classification-prediction'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14357953765
